modules/accounting/includes/functions/purchase-return.php

```diff
@@ -0,0 +1,531 @@
+        }
+    }
+
+    if ( ! empty( $args['type'] ) ) {
+        $where .= " AND voucher.type = '{$args['start_date']}'";
+    }
+
+    if ( - 1 !== $args['number'] ) {
+        $limit = "LIMIT {$args['number']} OFFSET {$args['offset']}";
+    }
+
+    $sql = 'SELECT';
+
+    if ( $args['count'] ) {
+        $sql .= ' COUNT( DISTINCT voucher.id ) AS total_number';
+    } else {
+        $sql .= ' voucher.id,
+            voucher.type,
+            purchase.invoice_id as purchase_voucher_no,
+            purchase.vendor_id as vendor_id,
+            purchase.vendor_name AS vendor_name,
+            purchase.trn_date AS bill_trn_date,
+            purchase.amount,
+            purchase.tax,
+            purchase.discount,
+            purchase.status AS purchase_status ';
+    }
+
+    $sql .= " FROM {$wpdb->prefix}erp_acct_voucher_no AS voucher
+         JOIN {$wpdb->prefix}erp_acct_purchase_return AS purchase ON purchase.voucher_no = voucher.id
+        {$where} GROUP BY voucher.id ORDER BY voucher.id {$args['order']} {$limit}";
+
+    if ( $args['count'] ) {
+        $wpdb->get_results( $sql );
+
+        return $wpdb->num_rows;
+    }
+
+    // error_log(print_r($sql, true));
+    return $wpdb->get_results( $sql, ARRAY_A );
+}
+
+
+/**
+ * Get an purchase invoice for return
+ *
+ * @param $invoice_no
+ *
+ * @return mixed
+ */
+function erp_acct_get_invoice_for_purchase_return( $invoice_no ) {
+    global $wpdb;
+
+    $sql = $wpdb->prepare(
+        "SELECT
+
+    voucher.editable,
+    purchase.id,
+    purchase.voucher_no,
+    purchase.vendor_id,
+    purchase.trn_date,
+    purchase.due_date,
+    purchase.amount,
+    purchase.vendor_name,
+    purchase.billing_address,
+    purchase.ref,
+    purchase.status,
+    purchase.purchase_order,
+    purchase.attachments,
+    purchase.particulars,
+    purchase.created_at,
+    purchase_acc_detail.purchase_no,
+    purchase_acc_detail.debit,
+    purchase_acc_detail.credit,
+    purchase_return.amount as return_amount,
+    purchase_return.discount as return_discount,
+    purchase_return.tax as return_tax,
+    purchase_return.reason as return_reason,
+    purchase_return.comments as return_comments,
+    purchase_return.trn_date as return_trn_date
+
+    FROM {$wpdb->prefix}erp_acct_purchase AS purchase
+    LEFT JOIN {$wpdb->prefix}erp_acct_voucher_no as voucher ON purchase.voucher_no = voucher.id
+    LEFT JOIN {$wpdb->prefix}erp_acct_purchase_account_details AS purchase_acc_detail ON purchase.voucher_no = purchase_acc_detail.trn_no
+    LEFT JOIN {$wpdb->prefix}erp_acct_purchase_return as purchase_return ON purchase.id = purchase_return.invoice_id
+    WHERE purchase.voucher_no = %d",
+        $invoice_no
+    );
+
+    $row = $wpdb->get_row( $sql, ARRAY_A );
+
+    $row['line_items']  = erp_acct_format_invoice_line_items_for_return( $invoice_no );
+    $row['tax_rate_id'] = erp_acct_get_default_tax_rate_name_id();
+
+    $row['attachments'] = unserialize( $row['attachments'] );
```

Comment:
You may consider using `maybe_unserialize` which initially checks if the data is serialized.

Reply:
ok